src/components/Modal/Modal.jsx

```diff
@@ -1,10 +1,110 @@
-import React from 'react';
+import React, { useState, useEffect } from "react";
+import { AiFillCloseCircle, AiFillCheckSquare } from "react-icons/ai";
+import { BsPlusLg, BsFillPlusCircleFill } from "react-icons/bs";
+import {
+  useAuth,
+  useVideo,
+  addNewPlaylist,
+  addVideoToPlaylist,
+  getAllPlaylist,
+  isVideoPresentInPlaylist,
+  removeVideoFromPlaylist
+} from "../../context";
 import "./Modal.css";
+import { ToastMsg } from "../ToastMsg/ToastMsg";
+
+const Modal = () => {
+  const { authState } = useAuth();
+  const { isLoggedIn } = authState;
+  const { videoState, videoDispatch, setShowModal, video } = useVideo();
+  const { playlists } = videoState;
+  const [playlistName, setPlaylistName] = useState("");
+
+  const handleChange = (e) => {
+    setPlaylistName(e.target.value);
+  };
+
+  const handleAddPlaylist = () => {
+    if (playlistName !== "") {
+      addNewPlaylist(
+        isLoggedIn,
+        {
+          title: playlistName,
+          description: "New Playlist",
+        },
+        videoDispatch
+      );
+    } else {
+      ToastMsg("Enter Playlist Title", "warning");
+    }
+    setPlaylistName("");
+  };
+
+  useEffect(() => {
+    getAllPlaylist(isLoggedIn, videoDispatch);
+  }, [playlists]);
 
-const Modal =() =>{
   return (
-    <div>Modal</div>
-  )
-}
+    <div className="modal-box">
```

Comment:
Consider using React Portal here for better support. Going with the current approach can hinder where your Modal appears